java/vector/src/main/java/org/apache/arrow/vector/TinyIntVector.java

```diff
@@ -370,6 +371,12 @@ public TransferPair makeTransferPair(ValueVector to) {
     return new TransferImpl((TinyIntVector) to);
   }
 
+  @Override
+  public void setEncodedValue(int index, int value) {
+    Preconditions.checkArgument(value <= Byte.MAX_VALUE, "value is overflow:" + value);
```

Comment:
```suggestion
    Preconditions.checkArgument(value <= Byte.MAX_VALUE, "value is overflow: %s" , value);
```
Otherwise you pay the cost of string concatenation on every iteration (same applies below)

Reply:
fixed.